src/go/rpk/pkg/cli/cmd/debug/bundle.go

```diff
@@ -0,0 +1,842 @@
+		}
+		bs, err := json.Marshal(res)
+		if err != nil {
+			return fmt.Errorf("couldn't encode resource usage metrics: %w", err)
+		}
+		return writeFileToZip(ps, "resource-usage.json", bs)
+	}
+}
+
+// Queries 'pool.ntp.org' and writes a file with the reported RTT, time & precision.
+func saveNTPDrift(ps *stepParams) step {
+	return func() error {
+		host := "pool.ntp.org"
+
+		response, err := ntp.Query(host)
+		if err != nil {
+			return fmt.Errorf("error querying '%s': %w", host, err)
+		}
+
+		result := struct {
+			Host            string    `json:"host"`
+			RoundTripTimeMs int64     `json:"roundTripTimeMs"`
+			Time            time.Time `json:"time"`
+			PrecisionMs     int64     `json:"precisionMs"`
+		}{
+			Host:            host,
+			RoundTripTimeMs: response.RTT.Milliseconds(),
+			Time:            response.Time,
+			PrecisionMs:     response.Precision.Milliseconds(),
+		}
+
+		marshalled, err := json.Marshal(result)
+		if err != nil {
+			return fmt.Errorf("couldn't marshal the NTP response: %w", err)
+		}
+
+		return writeFileToZip(
+			ps,
+			"ntp.txt",
+			marshalled,
+		)
+	}
+}
+
+func saveSyslog(ps *stepParams) step {
+	return func() error {
+		entries, err := syslog.ReadAll()
+		if err != nil {
+			return err
+		}
+		return writeFileToZip(ps, "syslog.txt", entries)
+	}
+}
+
+// Queries the given admin API address for prometheus metrics.
+func savePrometheusMetrics(ps *stepParams, api *admin.AdminAPI) step {
+	return func() error {
+		raw, err := api.PrometheusMetrics()
+		if err != nil {
+			return fmt.Errorf("unable to fetch metrics from the admin API: %w", err)
+		}
+		return writeFileToZip(ps, "prometheus-metrics.txt", raw)
+	}
+}
+
+// Saves the output of `dig`
+func saveDNSData(ps *stepParams) step {
+	return func() error {
+		return writeCommandOutputToZip(ps, "dig.txt", "dig")
+	}
+}
+
+// Saves the disk usage total within redpanda's data directory.
+func saveDiskUsage(ps *stepParams, conf *config.Config) step {
+	return func() error {
+		return writeCommandOutputToZip(
+			ps,
+			"du.txt",
+			"du", "-h", conf.Redpanda.Directory,
+		)
+	}
+}
+
+// TODO: What if running inside a container/ k8s?
+// Writes the journald redpanda logs, if available, to the bundle.
+func saveLogs(ps *stepParams, since, until string, logsLimitBytes int) step {
+	return func() error {
+		args := []string{"--no-pager", "-u", "redpanda"}
+		if since != "" {
+			args = append(args, "--since", since)
+		}
+		if until != "" {
+			args = append(args, "--until", until)
+		}
+		return writeCommandOutputToZipLimit(
+			ps,
+			"redpanda.log",
+			logsLimitBytes,
+			"journalctl",
+			args...,
+		)
+	}
+}
+
+// Saves the output of `ss`
+func saveSocketData(ps *stepParams) step {
+	return func() error {
+		return writeCommandOutputToZip(ps, "ss.txt", "ss")
+	}
+}
+
+// Saves the output of `top`
+func saveTopOutput(ps *stepParams) step {
+	return func() error {
+		return writeCommandOutputToZip(
+			ps,
+			"top.txt",
+			"top", "-n", "10", "-H",
```

Comment:
`-n 100` may be a bit better. If redpanda is stuck, it may not be using much CPU, but maybe it has like 100G of memory in use which would be significant for us to know.

Reply:
-n is the number of iterations, not lines 🙏🏽